### ISSUE TITLE
Trying to infer the `batch_size` from an ambiguous collection

### ISSUE THREAD

**Issue by je-santos** [PyTorchLightning/pytorch-lightning#12071]
Dear lightning friends,

I have a dataloader that has data in a specific format (it outputs a couple lists of tensors). 

o. lightning is unable to tell what's the batch size, so I'm wondering should I manually tell the logger that the batch_size is 1 or should it be 1*accumulate_grad_batches?

o. Just out of curiosity, lightning seems to think my batch size is between 14 and 17, where is it getting that information from?

Thanks for your great work!

best,
Javier

**Comment by rohitgr7** [comment]
does your dataset returns any str dtype? because PyTorch creates a list of strings in such a case, which is not a tensor so Lightning doesn't interpret the batch_size from something that is not a tensor.

You can specify your batch_size using `self.log(..., batch_size=1)`. This should not include anything related to gradient accumulation factor.

**Comment by je-santos** [comment]
That makes sense. I recently modified my data loader to return a string.
This was helpful, many thanks.

**Comment by adisingh50** [comment]
I have a similar problem. Specifically, my `__getitem__` function from my `Dataset` class specifically returns `Tuple[str, int, np.ndarray]`. I require the `str` and `int` to generate the input into my neural network, and the `np.ndarray` serves as ground truth label for said input. Anyways, in which function call am I supposed to insert: `self.log(..., batch_size=1)` in order for the lightning training script to correctly determine the total number of batches I need for training.

**Comment by rohitgr7** [comment]
> to correctly determine the total number of batches I need for training.

not the number of batches but the batch_size of the current batch being processed.
